client/src/dataset/addtoproject/DatasetAdd-types.ts

```diff
@@ -0,0 +1,34 @@
+import { IDataset } from "../../features/project/Project";
+
+interface AddDatasetDataset extends IDataset {
```

Comment:
Exports are more readable when done at the definition site.

```suggestion
export interface AddDatasetDataset extends IDataset {
```

Reply:
The comment above applies here as well.